extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -165,6 +177,15 @@ class ConnectorsManager {
                     kafkaConnectHost: this._kafkaConnectHost,
                     kafkaConnectPort: this._kafkaConnectPort,
                 });
```

Comment:
* what you compute here is not the number of retained buckets (i.e. the ones which are unexpectedly listened to), but only the deviation from the max number of buckets (which may not be an issue it itself)
* the number of "extra" buckets is not `buckets.length`, but the sum of extra buckets for each connector

---

typically, if the platform was setup to listen more than 10 buckets:
* after upgrade to mongo 6 & al, indeed we will have one or more connector which listen to more than 1 bucket --> not very important, just not the "new" nominal state
* eventually, the user stop listening to some bucket --> now we will have "real" inefficiency, as we will keep listening to that bucket

--> The first info may be represented as the number of connector which have more buckets than expected ; or the "total" number of buckets which do not fit in the list (e.g. sum(buckets.length - _maximumBucketsPerConnector) for every connector)
--> The seocnd info is the number entries in retainBuckets map at the end of the periodic update of all connectors

Reply:
I updated to add these two metrics